### PR TITLE
Feature/create transaction footer swiftui

### PR DESCRIPTION
**Pull request by acassiovilasboas**
## Description

- added transactionFooter in SwiftUI

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Screenshots (if appropriate):

https://github.com/ocean-ds/ocean-ios/assets/47704536/fe4bdb00-0249-4998-a7ab-490ac2b089b6

Simulator Screenshot - iPhone 15 - 2024-05-29 at 08 21 56

## Checklist:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have made corresponding changes to the documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All new and existing tests passed

### CODE REVIEW

**Review by vromeiro — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/TransactionFooter/OceanSwiftUI+TransactionFooter.swift [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+    // MARK: Parameters
+
+    public class TransactionFooterParameters: ObservableObject {
+        @Published public var items: [ItemModel]
+        @Published public var primaryButton: ButtonParameters?
+        @Published public var secondaryButton: ButtonParameters?
+        @Published public var buttonOrientation: ButtonOrientation
+        @Published public var showSkeleton: Bool
+        @Published public var numberOfItemsToShowSkeleton: Int
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/TransactionFooter/OceanSwiftUI+TransactionFooter.swift:21]: Man, pra deixar no mesmo esquema do componente Skeleton, eu sugiro o nome skeletonLines

File: OceanComponents/Classes/Components SwiftUI/TransactionFooter/OceanSwiftUI+TransactionFooter.swift [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        public enum ButtonOrientation {
+            case horizontal
+            case vertical
+        }
+
+        public class ItemModel: ObservableObject, Identifiable {
+            @Published public var text: String
+            @Published public var value: String
+            @Published public var valueColor: UIColor
+            @Published public var valueIsBold: Bool
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/TransactionFooter/OceanSwiftUI+TransactionFooter.swift:52]: Eu sugiro usar o "is" no começo pra manter o padrão de variável booleana. Então eu sugiro somente um "isBold" ou "isBoldValue". Talvez até faça mais sentido um "bold" somente, sabe?

File: OceanComponents/Classes/Components SwiftUI/TransactionFooter/OceanSwiftUI+TransactionFooter.swift [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                if parameters.showSkeleton {
+                    getSkeletonView(itemsCount: parameters.numberOfItemsToShowSkeleton)
+                } else {
+                    ForEach(parameters.items.indices, id: \.self) { index in
+                        getItemView(item: parameters.items[index])
+                    }
+
+                    if parameters.buttonOrientation == .horizontal {
+                        HStack {
+                            getButtonsView()
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/TransactionFooter/OceanSwiftUI+TransactionFooter.swift:119]: Gostei da ideia do método retornar só os botões e se vai ser vertical ou não aqui em quem chama, muito boa ideia!

File: OceanComponents/Classes/Components UIKit/TransactionFooter/Ocean+TransactionFooterItemView.swift [hunk 4]
```diff
@@ -1,5 +1,5 @@
 //
-//  Ocean+TransectionFooterListItem.swift
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components UIKit/TransactionFooter/Ocean+TransactionFooterItemView.swift:2]: kkkkkkkkk

**Review by renanmassaroto — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/TransactionFooter/OceanSwiftUI+TransactionFooter.swift (hunk 2 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Components SwiftUI/TransactionFooter/OceanSwiftUI+TransactionFooter.swift:52]: Boa, era "bold" no componente em uiKit, aqui vou colocar o isBoldValue pois a propriedade altera a fonte do conteudo do "value".

File: OceanComponents/Classes/Components UIKit/TransactionFooter/Ocean+TransactionFooterItemView.swift (hunk 4 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Components UIKit/TransactionFooter/Ocean+TransactionFooterItemView.swift:2]: No names!

**Pull request merged**